### ISSUE TITLE
Handle non-200 responses

### ISSUE THREAD

**Issue by PeterBorah** [PeterBorah/vigor#3]
(I'm going to start adding issues for anything I need to do, but am not immediately working on. Feel free to beat me to the punch with a pull request!)

We need to handle non-200 responses. This should be a centralized thing, rather than something that each method that makes an API call handles itself. I'm in favor of raising exceptions, as a general pattern, but I don't have a design in mind yet.

**Comment by imogenkinsman** [comment]
(just a note)
List of Riot's HTTP error responses, according to documentation:

500 - internal server error
404 - summoner not found
401 - unauthorized
400 - bad request

Any non-200 response right now throws an error at `@revision_date = DateTime.strptime(data["revisionDate"].to_s,'%s')` right now, which is pretty unhelpful.

**Comment by imogenkinsman** [comment]
I'll have a push ready soon, writing tests.

Something interesting I noticed - I tried a bunch of id numbers to try to find one that's invalid (to test SummonerNotFound being raised). It looks like Riot just scrambles accounts that have been deleted/banned, and still returns the garbage data if you request it by id. If you try to use that id for other API requests (or try to find that id from the name), it'll return a 404. I've seen a bunch of people scramble data like this with wiped accounts on db's before (way less taxing than deleting a row), but it's weird that it's accessible at all through the API.

`"id":31640240,"name":"IS11f9c710874850c9b7b3c","profileIconId":9,"summonerLevel":1,"revisionDate":1371607602000,"revisionDateStr":"06/19/2013
        02:06 AM UTC"`
